src/libutil/sysutil.cpp

```diff
@@ -604,4 +616,60 @@ aligned_free(void* ptr)
 }
 
 
+
+std::string
+Sysutil::stacktrace()
+{
+#if BOOST_VERSION >= 106500
+    std::stringstream out;
+    out << boost::stacktrace::stacktrace();
```

Comment:
Seems like a shame to require memory allocation (however small it may be) inside the signal handler.

In what case do you think the user would want to access this string as memory instead of relying on the installed handler?

Reply:
Well, I don't consider it ok to install signal handlers as a library, unless explicitly told to do so. So at most I can guarantee that it's set for the standalone binaries (oiiotool, etc).

I can imagine all sorts of nonfatal situations in which it would be useful to print some kind of diagnostic or debugging message that can include the full stack trace, at times when no signal-triggering fault has occurred.

As far as the allocation... maybe? So what? It does seem to work, and it's a situation where we're about to crash anyway.